Realm/Realm/Thread Handover/ThreadSafeReference.cs

```diff
@@ -104,7 +106,8 @@ public static Object<T> Create<T>(T value)
         /// </param>
         /// <typeparam name="T">The type of the objects contained in the list.</typeparam>
         /// <returns>A <see cref="ThreadSafeReference"/> that can be passed to <see cref="Realm.ResolveReference{T}(List{T})"/> on a different thread.</returns>
-        public static List<T> Create<T>(IList<T> value) => new List<T>(value);
+        public static List<T> Create<T>(IList<T> value)
+            => new List<T>(Argument.EnsureType<RealmList<T>>(value, "value must be a managed List (i.e. property of a RealmObject).", nameof(value)));
```

Comment:
This change replaces an InvalidCastException that would have been thrown when we attempt to cast `value` to the correct type with an `ArgumentException` that contains more actionable message.